internal/controller/snapshot/snapshot_adapter_test.go

```diff
@@ -760,9 +761,107 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
+
+		BeforeEach(func() {
+			integrationTestScenarioInvalid = &v1beta2.IntegrationTestScenario{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "invalid-its",
+					Namespace: "default",
+
+					Labels: map[string]string{
+						"test.appstudio.openshift.io/optional": "false",
+					},
+				},
+				Spec: v1beta2.IntegrationTestScenarioSpec{
+					Application: "application-sample",
+					Params: []v1beta2.PipelineParameter{
+						{
+							Name:  "HERMETIC",
+							Value: "true",
+						},
+						{
+							Name:  "hermetic",
+							Value: "true",
+						},
+					},
+					ResolverRef: v1beta2.ResolverRef{
+						Resolver: "git",
+						Params: []v1beta2.ResolverParameter{
+							{
+								Name:  "url",
+								Value: "https://github.com/redhat-appstudio/integration-examples.git",
+							},
+							{
+								Name:  "revision",
+								Value: "main",
+							},
+							{
+								Name:  "pathInRepo",
+								Value: "pipelineruns/integration_pipelinerun_pass.yaml",
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, integrationTestScenarioInvalid)).Should(Succeed())
+			helpers.SetScenarioIntegrationStatusAsValid(integrationTestScenarioInvalid, "valid")
+
+			Eventually(func() error {
+				err := k8sClient.Get(ctx, types.NamespacedName{
+					Name:      hasSnapshot.Name,
+					Namespace: "default",
+				}, hasSnapshot)
+				return err
+			}, time.Second*10).ShouldNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(ctx, hasSnapshot)
+			Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+			err = k8sClient.Delete(ctx, integrationTestScenarioInvalid)
+			Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+		})
+
+		It("will requeue reconciliation with error due to duplicated params", func() {
+			var buf bytes.Buffer
+
+			log := helpers.IntegrationLogger{Logger: buflogr.NewWithBuffer(&buf)}
+			adapter = NewAdapter(ctx, hasSnapshot, hasApp, hasComp, log, loader.NewMockLoader(), k8sClient)
+
+			adapter.context = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ApplicationContextKey,
+					Resource:   hasApp,
+				},
+				{
+					ContextKey: loader.ComponentContextKey,
+					Resource:   hasComp,
+				},
+				{
+					ContextKey: loader.SnapshotContextKey,
+					Resource:   hasSnapshot,
+				},
+				{
+					ContextKey: loader.SnapshotComponentsContextKey,
+					Resource:   []applicationapiv1alpha1.Component{*hasComp},
+				},
+				{
+					ContextKey: loader.AllIntegrationTestScenariosContextKey,
+					Resource:   []v1beta2.IntegrationTestScenario{*integrationTestScenarioInvalid},
+				},
+				{
+					ContextKey: loader.RequiredIntegrationTestScenariosContextKey,
+					Resource:   []v1beta2.IntegrationTestScenario{*integrationTestScenarioInvalid},
+				},
+			})
+
+			result, err := adapter.EnsureIntegrationPipelineRunsExist()
+			Expect(result.RequeueRequest).To(BeTrue())
```

Comment:
I'm not sure why the unit test is able to create the Integration PLR even though the corresponding ITS is invalid as it contains duplicate params. I'd expect it to call the `controller.RequeueWithError(errsForPLRCreation)`

Reply:
I think that error is only caught with webhooks ([STONEINTG-876](https://issues.redhat.com/browse/STONEINTG-876)) so that error doesn't show up in unit tests.

Reply:
Oh. Thanks a lot for letting me know. Let me update the unit tests.

Reply:
Removed the unit tests as per the discussion on Slack. PTAL